### PR TITLE
Tator codewords now appear in red when seen by other tators/nukeops

### LINKED ISSUE #22613: Syndicate Codewords should appear in a color that stands out in chat, but only for the agents that have them.
[featurerequest] [qol]

Basically like this:

Syndie Scum 1 has the codewords "tequila, MoMMI, scrubber" and the responses "janitor, medbay, screwdriver"
Syndie Scum 1 and Syndie Scum 2 share the same codewords and responses, and see the same text effects in the chat.

>Syndie Scum 1 [Common] (Assistant) says, "Hey, barman! **Tequila** sunrise please!"
>(not long afterward)
>Syndie Scum 2 [Common] (Scientist) says, "Could we get a **janitor** to science, please?"

I'm not sure if there should be two different colors (one for codeword and one for respective response) or if all codewords and responses should share the same color? Either way, this is to make people attempt to use their codewords and responses more, and maybe coordinate between other Syndicate agents in more rounds? Saying your codeword would make other Syndicate agents also see the codeword with the different text color in the chat - which would out you to them. And Syndicate agents are not obligated to cooperate with each other in their nefarious antagging.

This would:
>A) Cause uncooperative Syndicate agents who hear you to know you're a fellow agent and possibly target you. You wouldn't know who they are in turn unless they follow up with the appropriate response, which carries about as much risk as you would expect… unless you're doing this intentionally for a fun round of counter-subterfuge.

>B) Cause cooperative Syndicate agents who hear you to know you're a fellow agent and try to establish contact with you. By dropping your codeword and fishing for a response, you could bait other agents into outing themselves to you and risking you working against them in turn. This doesn't stop them from being more subtle about _how_ they establish contact outside of dropping their response word, though.

Thoughts on this? I cannot code anything for shit, but I wanted to drop this here anyway so others may benefit from the idea.

**EDIT:** Also should the words _only_ be color-indicated when it is another Syndicate agent that speaks them in chat, or when _anybody_ speaks them in chat?

The possibilities I see from "anybody who speaks a codeword, regardless of if they are a Syndicate agent or not, has it show up in color for Syndicate agents who hear it (but not themselves if they are a non-traitor)":

>A) You as a traitorous faggot speak your codeword over comms. The respective response to your codeword is enough of a commonly-used word or phrase that you can get multiple other people using it at the same time, but each time the word still has the same color indication. A devious Syndicate traitor who sees your codeword in the chat knows you are a traitor now, and might have spoken the response mixed with, say, _the four **other** greyshits spouting the same thing in a clusterfuck_ to muddy the waters for you - if they even speak their response for you to catch it at all.

>B) Same as the above, but applying to the initial codewords themselves too. A common-enough codeword could be something plausibly said by any multiple people within close timing of each other.

This way it won't simply be "Double Agents II: Electric Boogaloo" unless you make sure you use your codewords/responses in intentional ways only obvious to other agents through the color indication.
If ONLY Syndicate agents have their codewords/responses color indicated to other Syndicate agents, then that could allow for interesting wordplays and strategies like using "medical bay" instead of "medbay" if "medbay" specifically is your codeword, to avoid having you say "medbay" show up in unique color indication to that paramedic running by you in the hallway (who is a Syndicate agent but you don't know that - and since they notice the specific color text thanks to you, _they_ know _your_ identity now.)

### PR DESCRIPTION
**Pull request by PaulRitter**
[qol]

This will be some major qol for the tator metaclub ~altough i'm still not quite happy with my current markup, i tough about making it a little bolder~

Closes #22613

Codephrases are red and **bold**
Coderesponses are red and _italic_

Roles which get their text highlighted: Traitor, Nukeops

image

:cl:
 * rscadd: Syndicate Codephrases/responses now get highlighted for other syndicates when spoken

### CODE REVIEW

File: code/game/say.dm [hunk 1]
```diff
@@ -126,6 +126,15 @@ var/list/freqtoname = list(
 		pooled=1
 	else
 		say_testing(speech.speaker," We <i>do</i> understand this gentle\[wo\]man.")
+		//checking for syndie codephrases if person is a tator
+		if(istype(src, /mob/living))
+			var/mob/M = src
+			if(M.mind.special_role == TRAITOR || M.mind.special_role == NUKE_OP)
```
Comment (ShiftyRail) [inline code/game/say.dm]: `special_role` is a relic of old times
Use `M.mind.GetRole(TRAITOR)` or `M.mind.GetRole(NUKE_OP)`

**Comment by boy2mantwicethefam** [comment]
Real shit?

File: code/game/say.dm (hunk 1 above)
Comment (PaulRitter) [reply code/game/say.dm]: done.

**Comment by ShiftyRail** [comment]
Maybe make it bold for the code phrase and italic for the code answer ?

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
I want to say that this probably should be implemented into goonchat rather than serverside search and replace but it probably doesn't matter

**Comment by PaulRitter** [comment]
~no do tell, i'm happy to include it into goonchat, but this was the way i found saying stuff to be processed. just give me the .dm i'll try to work it in there~
Nevermind, im just gonna quite homoghost
>Someone could probably look at his cache and see the syndie words

**Comment by PaulRitter** [comment]
ready to merge, 100% tested and confirmed working

**Comment by SNilWillIllShill** [comment]
Can confirm, this is 100% tested and works. **MERGE IT**

**Comment by oversane** [comment]
MERGE NOW MERGE NOW

**Comment by PaulRitter** [comment]
For now the census seems to be to hold this merge since we wanna redo the codewords as a whole. Some seem to be saved fuckily so they aren't recognized by byond's replacetext(), which i use to style the text.

Also i am thinking about it only showing up as red when used in a sentence, since it can get pretty annoying and confusing for a tator when a codeword is something like "ill" or "scrub".

This will mean you'll have to wait a bit longer. If we would merge now you would have a system that would work most of the time, but would be super janky. Also the current codewords are shit and they need to be redone

**Comment by PaulRitter** [comment]
i removed the locations from the wordpool, all that codeword reworking should be done in another pr and should not hold back the highlighting feature, especially since apparently its quite wanted.

ready to merge

**Review by ShiftyRail — APPROVED** [review]

**Pull request merged**